src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -62,7 +122,23 @@ public boolean equals(Object other) {
         }
 
         DeleteCommand otherDeleteCommand = (DeleteCommand) other;
-        return targetIndex.equals(otherDeleteCommand.targetIndex);
+
+        if (targetIndex == null && otherDeleteCommand.targetIndex == null
+                && predicate == null && otherDeleteCommand.predicate == null) {
+            return true;
+        }
+
+        if (targetIndex != null && otherDeleteCommand.targetIndex != null
+                && predicate == null && otherDeleteCommand.predicate == null) {
+            return targetIndex.equals(otherDeleteCommand.targetIndex);
+        }
+
+        if (targetIndex == null && otherDeleteCommand.targetIndex == null
+                && predicate != null && otherDeleteCommand.predicate != null) {
+            return predicate.equals(otherDeleteCommand.predicate);
+        }
+
+        return false;
```

Comment:
the if statement here is a bit confusing, would be better if can add comments on this or make the code clearer like this
```
 // Both commands have null fields
    boolean bothHaveNullTargets = targetIndex == null && otherDeleteCommand.targetIndex == null;
    boolean bothHaveNullPredicates = predicate == null && otherDeleteCommand.predicate == null;
    
    // Both commands have non-null fields
    boolean bothHaveTargets = targetIndex != null && otherDeleteCommand.targetIndex != null;
    boolean bothHavePredicates = predicate != null && otherDeleteCommand.predicate != null;
    
    // Case 1: Both have null targetIndex and null predicate
    if (bothHaveNullTargets && bothHaveNullPredicates) {
        return true;
    }
    
    // Case 2: Both have targetIndex but null predicate
    if (bothHaveTargets && bothHaveNullPredicates) {
        return targetIndex.equals(otherDeleteCommand.targetIndex);
    }
    
    // Case 3: Both have null targetIndex but have predicate
    if (bothHaveNullTargets && bothHavePredicates) {
        return predicate.equals(otherDeleteCommand.predicate);
    }
    
    // All other cases are false
    return false;
```

Reply:
Okay, will make the changes. Thank you!